pkg/k8s/init.go

```diff
@@ -18,14 +18,72 @@ package k8s
 import (
 	"fmt"
 	"os"
+	"time"
 
+	"github.com/cilium/cilium/pkg/backoff"
 	"github.com/cilium/cilium/pkg/logging/logfields"
 	"github.com/cilium/cilium/pkg/node"
+	"github.com/cilium/cilium/pkg/option"
 
 	go_version "github.com/hashicorp/go-version"
 	"github.com/sirupsen/logrus"
 )
 
+const (
+	nodeRetrievalMaxRetries = 15
+)
+
+func waitForNodeInformation(nodeName string) *node.Node {
+	backoff := backoff.Exponential{
```

Comment:
Out of curiosity, couldn't this be a controller?

Reply:
We do not support for this value to ever change so I'm not sure if there is a value in this being a controller. We want to block bootstrapping.